src/components/ammo-constraint.js

```diff
@@ -0,0 +1,141 @@
+/* global Ammo */
+
+module.exports = AFRAME.registerComponent("ammo-constraint", {
+  multiple: true,
+
+  schema: {
+    // Type of constraint.
+    type: { default: "lock", oneOf: ["lock", "fixed", "spring", "slider", "hinge", "coneTwist", "pointToPoint"] },
+
+    // Target (other) body for the constraint.
+    target: { type: "selector" },
+
+    // TODO: Wake up bodies when connected.
+    wakeUpBodies: { default: true },
+
+    // Offset of the hinge or point-to-point constraint, defined locally in the body.
+    pivot: { type: "vec3" },
+    targetPivot: { type: "vec3" },
+
+    // An axis that each body can rotate around, defined locally to that body.
+    axis: { type: "vec3", default: { x: 0, y: 0, z: 1 } },
+    targetAxis: { type: "vec3", default: { x: 0, y: 0, z: 1 } }
+  },
+
+  init: function() {
+    this.system = this.el.sceneEl.systems.physics;
+    this.constraint = null;
+  },
+
+  remove: function() {
+    if (!this.constraint) return;
+
+    this.system.removeConstraint(this.constraint);
+    this.constraint = null;
+  },
+
+  update: function() {
+    var el = this.el,
+      data = this.data;
+
+    this.remove();
+
+    if (!el.body || !data.target.body) {
+      (el.body ? data.target : el).addEventListener("body-loaded", this.update.bind(this, {}));
```

Comment:
Consider making this a single-use event listener with the `once` option: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Syntax. Ideally this listener would get cleaned up if the component is removed, too, but the library isn't currently too careful about that sort of thin.